api/controllers/MerchStoreController.ts

```diff
@@ -158,14 +159,27 @@ export class MerchStoreController {
 
   @UseBefore(UserAuthentication)
   @Post('/item/picture/:uuid')
-  async updateMerchPhoto(@UploadedFile('image',
+  async createMerchItemPhoto(@UploadedFile('image',
     { options: StorageService.getFileOptions(MediaType.MERCH_PHOTO) }) file: File,
     @Params() params: UuidParam,
-    @AuthenticatedUser() user: UserModel): Promise<UpdateMerchPhotoResponse> {
+    @AuthenticatedUser() user: UserModel): Promise<CreateMerchPhotoResponse> {
     if (!PermissionsService.canEditMerchStore(user)) throw new ForbiddenError();
-    const picture = await this.storageService.upload(file, MediaType.MERCH_PHOTO, params.uuid);
-    const item = await this.merchStoreService.editItem(params.uuid, { picture });
-    return { error: null, item };
+
+    // generate a random string for the picture url
+    const randomID = StorageService.getRandomString();
+    const picture = await this.storageService.uploadToFolder(file, MediaType.MERCH_PHOTO, randomID, params.uuid);
+    const photo = await this.merchStoreService.createItemPhoto(params.uuid, { picture });
```

Comment:
See my comment above on if we actually need `getRandomString()`, but if we don't, you can just set `randomId` equal to `uuid.v4()` to generate a random uuid.

Also nit: `randomId` isn't too useful of a variable name especially in the context of what it's used for - something like `fileName` or `uniqueFileName` might make more sense